pkgs/applications/virtualization/singularity/generic.nix

```diff
@@ -147,7 +152,7 @@ in
     openssl
     squashfsTools # Required at build time by SingularityCE
   ]
-  ++ lib.optional enableNvidiaContainerCli nvidia-docker
+  ++ lib.optional (enableNvidiaContainerCli && !supports-cdi) nvidia-docker
```

Comment:
`singularity`/`apptainer` only need the `libnvidia-docker`

Reply:
Just added the new condition to keep the same behavior.